e2e/helm_client.go

```diff
@@ -0,0 +1,286 @@
+	arg := make([]string, 0, 6)
+	arg = append(arg, "rollback", releaseName, strconv.Itoa(revision), "--tiller-namespace", m.Namespace)
+	_, err := m.executeUsingHelm(arg...)
+	return err
+}
+
+func (m *BinaryHelmManager) executeUsingHelmInNamespace(arg ...string) (string, error) {
+	arg = append(arg, "--namespace", m.Namespace, "--tiller-namespace", m.Namespace)
+	return m.executeUsingHelm(arg...)
+}
+
+func (m *BinaryHelmManager) executeUsingHelm(arg ...string) (string, error) {
+	if m.TillerHost != "" {
+		arg = append(arg, "--host", m.TillerHost)
+	}
+	return m.executeUsingBinary(m.HelmBin, arg...)
+}
+
+func (m *BinaryHelmManager) executeUsingBinary(binary string, arg ...string) (string, error) {
+	cmd := exec.Command(binary, arg...)
+	Logf("Running command %+v\n", cmd.Args)
+	stdout, err := cmd.Output()
+	if err != nil {
+		switch err.(type) {
+		case *exec.ExitError:
+			stderr := err.(*exec.ExitError)
+			Logf("Command %+v, Err %s\n", cmd.Args, stderr.Stderr)
+		case *exec.Error:
+			Logf("Command %+v, Err %s\n", cmd.Args, err)
+		}
+		return "", err
+	}
+	return string(stdout), nil
+}
+
+func (m *BinaryHelmManager) executeCommandWithValues(releaseName, command string, values map[string]string) (string, error) {
+	arg := make([]string, 0, 8)
+	arg = append(arg, command, releaseName)
+	if len(values) > 0 {
+		var b bytes.Buffer
+		for key, val := range values {
+			b.WriteString(key)
+			b.WriteString("=")
+			b.WriteString(val)
+			b.WriteString(",")
+		}
+		arg = append(arg, "--set", b.String())
+	}
+	return m.executeUsingHelmInNamespace(arg...)
+}
+
+func (m *BinaryHelmManager) InstallServiceAccounts() error {
+	objects := strings.Replace(serviceAccountTemplate, "TILLER_NAMESPACE", m.Namespace, -1)
+
+	f, err := ioutil.TempFile("", m.Namespace)
+	if err != nil {
+		Logf("Failed creating tempfile: %s", err)
+		return err
+	}
+
+	f.WriteString(objects)
+	f.Sync()
+
+	_, err = m.executeUsingBinary("kubectl", "create", "-f", f.Name())
+	return err
+}
+
+func regexpKeyFromStructuredOutput(key, output string) string {
+	r := regexp.MustCompile(fmt.Sprintf("%v:[[:space:]]*(.*)", key))
+	// key will be captured in group with index 1
+	result := r.FindStringSubmatch(output)
+	if len(result) < 2 {
+		return ""
+	}
+	return result[1]
+}
+
+func getNameFromHelmOutput(output string) string {
+	return regexpKeyFromStructuredOutput("NAME", output)
+}
+
+func getStatusFromHelmOutput(output string) string {
+	return regexpKeyFromStructuredOutput("STATUS", output)
+}
+
+func waitTillerPod(clientset kubernetes.Interface, namespace string) {
+	Eventually(func() bool {
+		pods, err := clientset.Core().Pods(namespace).List(metav1.ListOptions{})
+		if err != nil {
+			return false
+		}
+		for _, pod := range pods.Items {
+			if !strings.Contains(pod.Name, "tiller") {
+				continue
+			}
+			Logf("Found tiller pod. Phase %v\n", pod.Status.Phase)
+			if pod.Status.Phase != v1.PodRunning {
```

Comment:
I would recommend replacing this ready logic with `pod.IsPodReady` from "k8s.io/kubernetes/pkg/api/v1/pod" like we use in wait functionality

Reply:
Will look into it.

Reply:
Good tip, thanks.

Reply:
After all, it would require casting client-go types to k8s types, which I think will take as much code as current status checks. `IsPodReady` is not available in client-go, so I think I will pass on this one. Please let me know if you insist on having it.

Reply:
Nope, no problem at all. Now I kind of want to add this to `client-go` though 😄